blob/gcsblob/gcsblob.go

```diff
@@ -130,9 +132,12 @@ func (b *bucket) ListPaged(ctx context.Context, opt *driver.ListOptions) (*drive
 			return nil, err
 		}
 	}
+	pageSize := opt.PageSize
+	if pageSize == 0 {
+		pageSize = defaultPageSize
+	}
 	iter := bkt.Objects(ctx, query)
-	pager := iterator.NewPager(iter, opt.PageSize, string(opt.PageToken))
-
+	pager := iterator.NewPager(iter, pageSize, string(opt.PageToken))
```

Comment:
This is overkill. Use the client's iterator, but set the page size with https://godoc.org/cloud.google.com/go/storage#ObjectIterator.PageInfo.

Reply:
(May make more sense in a separate PR.)

Reply:
I took a look, and yes, I could set `iter.PageInfo.Token` and `iter.PageInfo.MaxSize`, and then use `iter.Next()` until `iter.PageInfo.Remaining == 0`. That's what you're suggesting, right? I'm not sure that's simpler than what I have here.... do you think it's simpler, or is there some other kind of overhead to using `NewPager`?

Reply:
I see what's going on here. The driver API is at the right level, but you're using a high-level client (c.g.c/go/storage) that's already hiding pagination for you. As a result, `gcsblob.bucket.ListPaged` is doing a lot of extra work on each page.

I'll need to think about what the right thing to do here is. But I don't think it needs to block this PR.

Reply:
OK, will leave it like this for now.